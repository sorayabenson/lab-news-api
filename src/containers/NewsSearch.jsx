import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Spinner from '../components/Spinner';
import ArticleList from '../components/article/ArticleList';
import Search from '../components/article/Search';
import Header from '../components/Header';
import { getThePaper } from '../services/newsApi';
import '../components/style.css';

export default class NewsSearch extends Component {
    state = {
        loading: true,
        articles: []
    }

    async componentDidMount() {
        const articles = await getThePaper();
        
        this.setState({
            loading: false,
            articles: articles
        })
    }

    render() {
        const { loading, articles } = this.state;

        if (loading) return <Spinner />

        return (
            <>
                <Header />
                <Search />
                <ArticleList articles={articles}/>
            </>
        )
    }
}
