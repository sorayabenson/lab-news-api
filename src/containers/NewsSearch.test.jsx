import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NewsSearch from './NewsSearch';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

require('dotenv').config();

const server = setupServer(
  rest.get(
    `https://fauxnewsapi.com/v1/search`,
    (req, res, ctx) => {
      const query = req.url.searchParams
      const q = query.get('ghosts')
      return res(
        ctx.json([
          {
            title: 'Ghosts, are they you?',
            author: 'Mr. Boo',
            description: 'A study on past, present, and future ghosts',
            url: 'www.ghost.com',
            source: 'The Washington Ghost'
          }
        ])
      )
    })
)

describe('NewsSearch container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('renders App', async () => {
    render(<NewsSearch />);

    screen.getByLabelText('loading')

    const ul = await screen.findByRole('list', { name: 'articles' });
    expect(ul).not.toBeEmptyDOMElement();
    expect(ul).toMatchSnapshot();

    const input = await screen.findByLabelText('article search');
    userEvent.type(input, 'ghosts');
    expect(input).toMatchSnapshot();

    const submitButton = await screen.findByRole('button', { name: 'search-articles' })
    userEvent.click(submitButton);
    expect(submitButton).toMatchSnapshot();

    const article = await screen.findAllByRole('listitem', { name: 'article' })
    expect(article).toMatchSnapshot();
  });
});
