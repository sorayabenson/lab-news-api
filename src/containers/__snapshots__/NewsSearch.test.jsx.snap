// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsSearch container renders App 1`] = `
<ul
  aria-label="articles"
>
  <li
    aria-label="article"
  >
    <a
      href="https://lifehacker.com/what-the-hell-happened-to-aukeys-products-on-amazon-1846870233"
    >
      <h3>
        What the Hell Happened to Aukey's Products on Amazon?
      </h3>
      <h4>
        Aukey makes good chargers, or so the many reviews and roundups of them lead me to believe. However, if you’ve been Amazon shopping lately, you might be surprised to find Aukey’s portfolio of products mysteriously missing.Read more...
      </h4>
      <p
        class="author"
      >
        David Murphy
      </p>
      <p
        class="source"
      >
        Lifehacker.com
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="http://techcrunch.com/2021/04/19/european-vc-soars-in-q1/"
    >
      <h3>
        European VC soars in Q1
      </h3>
      <h4>
        A stunning first quarter in venture capital funding was not restricted to the United States; Europe also had one hell of a start to the year.
      </h4>
      <p
        class="author"
      >
        Anna Heim
      </p>
      <p
        class="source"
      >
        TechCrunch
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://www.nytimes.com/2021/04/20/arts/music/jim-steinman-bat-out-of-hell-songwriter-dies-at-73.html"
    >
      <h3>
        Jim Steinman, ‘Bat Out of Hell’ Songwriter, Dies at 73
      </h3>
      <h4>
        The rocker Meat Loaf’s interpretations of Mr. Steinman’s songs became one of the biggest-selling albums of all time.
      </h4>
      <p
        class="author"
      >
        Neil Genzlinger
      </p>
      <p
        class="source"
      >
        New York Times
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://io9.gizmodo.com/what-the-hell-is-batwoman-doing-to-kate-kane-1846896121"
    >
      <h3>
        What the Hell Is Batwoman Doing to Kate Kane?
      </h3>
      <h4>
        When Ruby Rose announced she was leaving Batwoman last May, the show had a problem—several problems, in fact. Rose was of course the star and lead of the series, but every single part of the series was built solely around Kate Kane, so the CW’s writers needed…
      </h4>
      <p
        class="author"
      >
        Rob Bricken
      </p>
      <p
        class="source"
      >
        Gizmodo.com
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://gizmodo.com/holy-hell-peloton-is-recalling-both-its-treadmills-1846827897"
    >
      <h3>
        Holy Hell, Peloton Is Recalling Both Its Treadmills
      </h3>
      <h4>
        After a heated back-and-forth with the U.S. Consumer Product Safety Commission (CPSC), Peloton has finally decided to issue two separate voluntary recalls of its Tread and Tread+ treadmills.Read more...
      </h4>
      <p
        class="author"
      >
        Victoria Song
      </p>
      <p
        class="source"
      >
        Gizmodo.com
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://lifehacker.com/top-10-ways-to-tame-and-deal-with-your-allergies-1757499217"
    >
      <h3>
        10 Ways to Better Tame Your Spring Allergies
      </h3>
      <h4>
        Sneezing, wheezing, and otherwise feeling like hell: Allergies are a special kind of everyday torture. Although we don’t have a cure for your food or environmental allergies, we do have some proven tips to help you suffer less and minimize reactions. Here are…
      </h4>
      <p
        class="author"
      >
        Melanie Pinola
      </p>
      <p
        class="source"
      >
        Lifehacker.com
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://lifehacker.com/how-do-i-install-windows-10-without-all-those-extra-upd-1846790600"
    >
      <h3>
        How Do I Install Windows 10 Without All Those Extra Updates?
      </h3>
      <h4>
        Every Windows PC would benefit from a spring cleaning. Hell, it doesn’t even have to be spring. If you can’t remember the last time you saved all your critical data, wiped your drive, and reinstalled a fresh copy of the operating system, you’re due. Read more…
      </h4>
      <p
        class="author"
      >
        David Murphy
      </p>
      <p
        class="source"
      >
        Lifehacker.com
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://lifehacker.com/what-to-do-if-your-dog-suffers-from-separation-anxiety-1846859612"
    >
      <h3>
        What to Do If Your Dog Suffers From Separation Anxiety
      </h3>
      <h4>
        Nuzzles, cuddles, and walks were available in abundance as people were confined to their homes during the pandemic, holed up with their pets, who were blissfully unaware of the mortifying circumstances of the outside world. But now that the United States appe…
      </h4>
      <p
        class="author"
      >
        Sam Blum
      </p>
      <p
        class="source"
      >
        Lifehacker.com
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://lifehacker.com/how-to-prepare-for-your-post-vax-slutty-summer-1846740170"
    >
      <h3>
        How to Prepare for Your Post-Vax Slutty Summer
      </h3>
      <h4>
        Adults are, of course, allowed to bang other consenting adults, and as of this week, they’re also all allowed to get vaccinated against COVID-19. And after over a year of mandated—and necessary—isolation, everybody’s randy. Hell, everybody’s been randy, but t…
      </h4>
      <p
        class="author"
      >
        Lindsey Ellefson
      </p>
      <p
        class="source"
      >
        Lifehacker.com
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://www.nytimes.com/2021/04/21/realestate/visiting-a-garden-center-this-spring-be-strategic.html"
    >
      <h3>
        Visiting a Garden Center This Spring? Be Strategic.
      </h3>
      <h4>
        There’s a reason the gardening industry fondly refers to this time of year as the ‘100 days of hell.’ And the pandemic has only made it more challenging.
      </h4>
      <p
        class="author"
      >
        Margaret Roach
      </p>
      <p
        class="source"
      >
        New York Times
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://mashable.com/article/what-is-the-omegaverse-tiktok-tumblr-fanfiction/"
    >
      <h3>
        What the hell is the Omegaverse, and why is it all over TikTok?
      </h3>
      <h4>
        Welcome, dear reader, to the Omegaverse.
The phrase has been part of fanfiction communities for the better part of the last decade, but it gained mainstream traction on TikTok in the past month. The tag #Omegaverse has over 357 million views on TikTok, and co…
      </h4>
      <p
        class="author"
      >
        Morgan Sung
      </p>
      <p
        class="source"
      >
        Mashable
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://www.reuters.com/breakingviews/indias-pandemic-hell-could-be-purgatory-2021-04-23/"
    >
      <h3>
        India's pandemic hell could be purgatory - Reuters
      </h3>
      <h4>
        India has slender room to act if economic activity follows its medical infrastructure into a virus-induced state of collapse. So far there’s a wide gap between the two measures of health. Optimists hope antibodies and vaccines will cause the Covid-19 contagio…
      </h4>
      <p
        class="author"
      />
      <p
        class="source"
      >
        Reuters
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://offspring.lifehacker.com/what-are-your-go-to-parenting-mantras-1846788377"
    >
      <h3>
        What Are Your Go-to Parenting Mantras?
      </h3>
      <h4>
        So much of parenting is about the long game. We may be raising children, but we’re really raising adults—people who, in a surprisingly short amount of time, will be off on their own, living their own lives. And how we parent them as kids will at least partial…
      </h4>
      <p
        class="author"
      >
        Meghan Moravcik Walbert
      </p>
      <p
        class="source"
      >
        Lifehacker.com
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://io9.gizmodo.com/mass-effect-made-itself-on-virmire-1846815333"
    >
      <h3>
        Mass Effect Made Itself on Virmire
      </h3>
      <h4>
        Eden Prime. Ilos. The Suicide Mission. The Battle of London. A small apartment with one hell of a party on the Citadel. There are many moments and places in the Mass Effect universe to turn to as some of the most memorable, definitive pieces of its story. But…
      </h4>
      <p
        class="author"
      >
        James Whitbrook
      </p>
      <p
        class="source"
      >
        Gizmodo.com
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://www.engadget.com/high-hopes-balloon-carbon-capture-120057079.html"
    >
      <h3>
        High Hopes plans to extract atmospheric CO2 with hot air balloons
      </h3>
      <h4>
        Freezing CO2 is easier just below the stratosphere.
      </h4>
      <p
        class="author"
      >
        https://www.engadget.com/about/editors/daniel-cooper
      </p>
      <p
        class="source"
      >
        Engadget
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://www.reuters.com/article/us-haiti-kidnappings-insight-idUSKBN2CD12O"
    >
      <h3>
        'Descent into hell': Kidnapping explosion terrorizes Haiti - Reuters
      </h3>
      <h4>
        A wave of kidnappings is sweeping Haiti. But even in a country growing inured to horrific abductions, the case of five-year-old Olslina Janneus sparked outrage.
      </h4>
      <p
        class="author"
      >
        Andre Paultre, Sarah Marsh
      </p>
      <p
        class="source"
      >
        Reuters
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://www.reuters.com/world/americas/descent-into-hell-kidnapping-explosion-terrorizes-haiti-2021-04-26/"
    >
      <h3>
        'Descent into hell': Kidnapping explosion terrorizes Haiti - Reuters
      </h3>
      <h4>
        A wave of kidnappings is sweeping Haiti. But even in a country growing inured to horrific abductions, the case of five-year-old Olslina Janneus sparked outrage.
      </h4>
      <p
        class="author"
      >
        Sarah Marsh
      </p>
      <p
        class="source"
      >
        Reuters
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://hackaday.com/2021/05/03/silicon-carbide-chips-can-go-to-hell/"
    >
      <h3>
        Silicon Carbide Chips can go to Hell
      </h3>
      <h4>
        IEEE Spectrum had an interesting read about circuits using silicon carbide as a substrate. [Alan Mantooth] and colleagues say that circuits based on this or some other rugged technology will …read more
      </h4>
      <p
        class="author"
      >
        Al Williams
      </p>
      <p
        class="source"
      >
        Hackaday
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://www.bbc.com/news/entertainment-arts-56825060"
    >
      <h3>
        Bat Out Of Hell composer Jim Steinman dies - BBC News
      </h3>
      <h4>
        &lt;ol&gt;&lt;li&gt;Bat Out Of Hell composer Jim Steinman dies  BBC News
&lt;/li&gt;&lt;li&gt;Jim Steinman, hitmaker for Meat Loaf and Céline Dion, dead at 73  CBC.ca
&lt;/li&gt;&lt;li&gt;Jim Steinman, Rock Legend and Songwriter for Meat Loaf and Céline Dion, Dead at 73: Report  PEOPLE
&lt;/li&gt;…
      </h4>
      <p
        class="author"
      >
        https://www.facebook.com/bbcnews
      </p>
      <p
        class="source"
      >
        BBC News
      </p>
    </a>
  </li>
  <li
    aria-label="article"
  >
    <a
      href="https://www.reuters.com/article/ip-nirvana-copyright-idUSL1N2MM2H5"
    >
      <h3>
        IN BRIEF: Nirvana sued over use of Dante's Inferno art in logo - Reuters
      </h3>
      <h4>
        The heir of a British artist who drew a map of hell for a translation of Dante's "Inferno" sued the band Nirvana in Los Angeles federal court for copyright infringement, alleging it misused the image on its merchandise.
      </h4>
      <p
        class="author"
      >
        Blake Brittain
      </p>
      <p
        class="source"
      >
        Reuters
      </p>
    </a>
  </li>
</ul>
`;

exports[`NewsSearch container renders App 2`] = `
<input
  id="article-search"
  type="text"
  value="ghosts"
/>
`;

exports[`NewsSearch container renders App 3`] = `
<button
  aria-label="search-articles"
>
  submit
</button>
`;

exports[`NewsSearch container renders App 4`] = `
Array [
  <li
    aria-label="article"
  >
    <a
      href="https://mashable.com/uk/shopping/april-18-best-cyberghost-vpn-deal/"
    >
      <h3>
        This secure and spooky VPN is on sale for under £2 per month
      </h3>
      <h4>
        SAVE 82%: A three-year subscription to CyberGhost VPN is on sale for £1.99 per month as of April 18, and includes an extra three months for free.

It's difficult to decide on the best VPN for you, but CyberGhost VPN does stand out for a number of reasons. It'…
      </h4>
      <p
        class="author"
      >
        Joseph Green
      </p>
      <p
        class="source"
      >
        Mashable
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://mashable.com/video/netflix-ghost-lab-trailer/"
    >
      <h3>
        Netflix's 'Ghost Lab' trailer sees two doctors trying to find proof of an afterlife
      </h3>
      <h4>
        GHOST LAB. The title might sound like that of a kids' film, but Paween Purijitpanya's upcoming Netflix movie is actually a fun-sounding Thai horror about two doctors trying to prove the existence of ghosts.
The trailer sees Gla (Paris Intarakomalyasut) and We…
      </h4>
      <p
        class="author"
      >
        Sam Haysom
      </p>
      <p
        class="source"
      >
        Mashable
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://www.wired.com/story/time-travel-is-not-cheating/"
    >
      <h3>
        Time Travel in Video Games Isn’t Really Cheating
      </h3>
      <h4>
        Games like 'Animal Crossing' and 'Cozy Grove' are meant to be played every day, but who’s got that kind of time?
      </h4>
      <p
        class="author"
      >
        Swapna Krishna
      </p>
      <p
        class="source"
      >
        Wired
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://venturebeat.com/2021/04/28/sony-expects-an-even-better-year-for-playstation-5-first-party-games/"
    >
      <h3>
        Sony expects an even better year for PlayStation 5 first-party games
      </h3>
      <h4>
        PlayStation 5 first-party games may have an even better year than 2020, which gave fans Ghosts of Tsushima, Last of Us Part II, and more.
      </h4>
      <p
        class="author"
      >
        Jeff Grubb
      </p>
      <p
        class="source"
      >
        VentureBeat
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://io9.gizmodo.com/updates-from-willow-disenchanted-and-more-1846759785"
    >
      <h3>
        Updates From Willow, Disenchanted, and More
      </h3>
      <h4>
        Both of Disney's new Willow and Enchanted sequels have expanded their casts, with Amer Chadha-Patel in a new leading role for Willow's followup, and Maya Rudolph, Yvette Nicole Brown and Jayma Mays reprising their Enchanted roles in Disenchanted.
      </h4>
      <p
        class="author"
      >
        James Whitbrook and Gordon Jackson
      </p>
      <p
        class="source"
      >
        Gizmodo.com
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://lifehacker.com/whats-new-on-amazon-prime-video-in-may-2021-1846723609"
    >
      <h3>
        What's New on Amazon Prime Video in May 2021
      </h3>
      <h4>
        Amazon Prime Video’s May slate of original series is a study in literary contrasts. On the one hand, we’ve got The Underground Railroad (May 14), a prestige limited series produced by Oscar-winner Barry Jenkins (Twelve Years a Slave), adapting Colson Whitehea…
      </h4>
      <p
        class="author"
      >
        Joel Cunningham
      </p>
      <p
        class="source"
      >
        Lifehacker.com
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://time.com/5956839/derek-chauvin-justice-police-violence/"
    >
      <h3>
        The Derek Chauvin Verdict Is Haunted by the Ghosts of Those Who Found No Justice
      </h3>
      <h4>
        When Judge Peter Cahill read the verdict that Derek Chauvin was guilty on all three counts for the murder of George Floyd I imagined ghosts dancing around the courtroom. They leapt from chair to chair. Shouting, laughing, and crying all at once. They were the…
      </h4>
      <p
        class="author"
      >
        Eddie S. Glaude Jr.
      </p>
      <p
        class="source"
      >
        Time
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://www.polygon.com/22425451/warhammer-fest-announcements-reveals-40k-orks-gaunts-ghosts-age-of-sigmar-release-date-price"
    >
      <h3>
        Warhammer Fest: 40K Ork refresh, Dan Abnett’s Gaunt’s Ghosts in plastic, and more
      </h3>
      <h4>
        Warhammer Fest still has one day left, but already there’s been a boat load of announcements and new models. 40K’s Orks are getting a refresh, plus new models of Dan Abnett’s Gaunt’s Ghosts. Age of Sigmar news rounds up the update with even more set for the w…
      </h4>
      <p
        class="author"
      >
        Luke Shaw
      </p>
      <p
        class="source"
      >
        Polygon
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://lwn.net/Articles/414618/"
    >
      <h3>
        Ghosts of Unix past, part 3: Unfixable designs
      </h3>
      <h4>
        In the second installment of this series on
design patterns from Unix, Neil Brown examined flawed designs which have
been slowly fixed over time.  But are there design patterns which are so
broken that they simply cannot be fixed?  In this article, Neil argue…
      </h4>
      <p
        class="author"
      />
      <p
        class="source"
      >
        Lwn.net
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://www.rockpapershotgun.com/phasmophobias-ghosts-are-even-harder-better-faster-stronger-now"
    >
      <h3>
        Phasmophobia's ghosts are now even harder, better, faster, stronger
      </h3>
      <h4>
        Yes, I know it is now half a year past the prime month for spooky games but ghost sleuthing sim Phasmophobia has not let that stop its continual updates. So I will not let it stop me from talking about them either. As of a beta branch update last weekend, Pha…
      </h4>
      <p
        class="author"
      >
        Lauren Morton
      </p>
      <p
        class="source"
      >
        Rock Paper Shotgun
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://io9.gizmodo.com/updates-from-the-suicide-squad-space-jam-2-and-more-1846906078"
    >
      <h3>
        Updates From The Suicide Squad, Space Jam 2, and More
      </h3>
      <h4>
        DC’s Hawkman may have some cool tech in Black Adam. A new look at Adult Swim’s pickup of Tuca & Bertie. Billy Porter looking like a dream in Cinderella. Plus new looks at Legends of Tomorrow, Fear the Walking Dead, and more. Spoilers save us from this Monday!…
      </h4>
      <p
        class="author"
      >
        James Whitbrook and Gordon Jackson
      </p>
      <p
        class="source"
      >
        Gizmodo.com
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://www.vice.com/en/article/n7bb4m/best-rental-houses-in-the-middle-of-nowhere"
    >
      <h3>
        Want to Travel But Still Feel Weird? Book a Vacation House in the Middle of Nowhere
      </h3>
      <h4>
        From 440-acre Joshua Tree estates to $69-a-night Airstream trailers full of horny ghosts (probably), there’s something out there for everyone.
      </h4>
      <p
        class="author"
      >
        Mary Frances Knapp, Hilary Pollack, Ian Burke
      </p>
      <p
        class="source"
      >
        Vice News
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://kotaku.com/one-u-s-dollar-will-buy-you-unlimited-invincibility-in-1846772671"
    >
      <h3>
        One U.S. Dollar Will Buy You Unlimited Invincibility In Capcom Arcade Stadium
      </h3>
      <h4>
        Having a hard time with the classic games of Capcom Arcade Stadium? Well, the developers have just the thing for you: paid invincibility.Read more...
      </h4>
      <p
        class="author"
      >
        Ian Walker
      </p>
      <p
        class="source"
      >
        Kotaku
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://io9.gizmodo.com/updates-from-black-widow-obi-wan-kenobi-and-more-1846856001"
    >
      <h3>
        Updates From Black Widow, Obi-Wan Kenobi, and More
      </h3>
      <h4>
        David Harbour has teased more connections to Red Guardian's superteam, the Winter Guard, in the new Black Widow movie. Meanwhile, Ewan McGregor teases some costume changes for Obi-Wan Kenobi in the new Disney+ Star Wars streaming series.
      </h4>
      <p
        class="author"
      >
        James Whitbrook and Gordon Jackson
      </p>
      <p
        class="source"
      >
        Gizmodo.com
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://kotaku.com/huh-the-new-fable-runs-on-the-forza-engine-1846833080"
    >
      <h3>
        Huh, The New Fable Runs On The Forza Engine
      </h3>
      <h4>
        The next installment in the Fable franchise is being built with the ForzaTech engine, which as the name implies, is the foundation beneath Xbox’s popular Forza racing games.Read more...
      </h4>
      <p
        class="author"
      >
        Ian Walker
      </p>
      <p
        class="source"
      >
        Kotaku
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://news.yahoo.com/man-ghosts-chapel-hill-woman-215104288.html"
    >
      <h3>
        Man ghosts Chapel Hill woman who paid him for backyard work
      </h3>
      <h4>
        "Where are you? Are you showing up today?" A Chapel Hill woman's contractor took the money and ran...and this contractor pleaded guilty to doing something...
      </h4>
      <p
        class="author"
      >
        WTVD – Raleigh/Durham
      </p>
      <p
        class="source"
      >
        Yahoo Entertainment
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://www.entrepreneur.com/article/371377"
    >
      <h3>
        Strengthen your bond with your mom with these 5 books
      </h3>
      <h4>
        Make her feel special without leaving home.
      </h4>
      <p
        class="author"
      >
        Entrepreneur en Español
      </p>
      <p
        class="source"
      >
        Entrepreneur
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://www.cnet.com/news/hulu-may-2021-everything-new-coming-this-month/"
    >
      <h3>
        Hulu May 2021: Everything new coming this month - CNET
      </h3>
      <h4>
        Catch the premiere of Marvel's M.O.D.O.K. and season three of Shrill
      </h4>
      <p
        class="author"
      >
        Nicole Archer
      </p>
      <p
        class="source"
      >
        CNET
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://time.com/5961384/things-heard-and-seen-netflix/"
    >
      <h3>
        What to Know About the Real-Life Inspiration Behind Netflix’s Things Heard & Seen
      </h3>
      <h4>
        Here's what to know about Netflix's 'Things Heard & Seen,' which is based on Elizabeth Brundage's novel 'All Things Cease to Appear.'
      </h4>
      <p
        class="author"
      >
        Annabel Gutterman
      </p>
      <p
        class="source"
      >
        Time
      </p>
    </a>
  </li>,
  <li
    aria-label="article"
  >
    <a
      href="https://ca.style.yahoo.com/meghan-remarks-kate-could-scupper-014558545.html"
    >
      <h3>
        Meghan’s remarks about Kate could scupper any bid to heal rift – royal author - Yahoo Canada Shine On
      </h3>
      <h4>
        &lt;ol&gt;&lt;li&gt;Meghan’s remarks about Kate could scupper any bid to heal rift – royal author  Yahoo Canada Shine On
&lt;/li&gt;&lt;li&gt;Kate Middleton and Prince William Wedding Guest Recalls Bride's Arrival: 'Everything Just Roared'  PEOPLE.com
&lt;/li&gt;&lt;li&gt;Doubtful Prince Will…
      </h4>
      <p
        class="author"
      >
        Laura Elston
      </p>
      <p
        class="source"
      >
        Yahoo Entertainment
      </p>
    </a>
  </li>,
]
`;
